handler.go

```diff
@@ -27,39 +29,41 @@ type langHandler struct {
 	request      chan DocumentURI
 	command      []string
 	noLinterName bool
+	showAll      bool
 
 	rootURI string
 }
 
-func (h *langHandler) lint(uri DocumentURI) ([]Diagnostic, error) {
-	diagnostics := make([]Diagnostic, 0)
+func (h *langHandler) lint(uri DocumentURI) (map[string][]Diagnostic, error) {
+	diagnostics := make(map[string][]Diagnostic, 0)
 
-	path := uriToPath(string(uri))
-	dir, file := filepath.Split(path)
+	file := strings.TrimPrefix(string(uri), h.rootURI+string(os.PathSeparator))
 
 	//nolint:gosec
 	cmd := exec.Command(h.command[0], h.command[1:]...)
-	cmd.Dir = dir
+	cmd.Dir = uriToPath(h.rootURI)
 
 	b, err := cmd.Output()
 	if err == nil {
-		return diagnostics, nil
+		return nil, nil
 	}
 
 	var result GolangCILintResult
 	if err := json.Unmarshal(b, &result); err != nil {
-		return diagnostics, err
+		return nil, err
 	}
 
 	h.logger.DebugJSON("golangci-lint-langserver: result:", result)
 
 	for _, issue := range result.Issues {
 		issue := issue
 
-		if file != issue.Pos.Filename {
+		if !h.showAll && file != issue.Pos.Filename {
 			continue
 		}
 
+		uri := filepath.Join(h.rootURI, issue.Pos.Filename)
```

Comment:
Diagnostic is not displayed because of this line.

expect: file:///foo/bar/buz
actual: /foo/bar/buz